systemtests/systemtestslib/utils.py

```diff
@@ -13,36 +14,51 @@
 CONNECTION_TIMEOUT = 600
 
 
-def get_sym_files(auth_token, url, start_page):
+def get_sym_files(baseurl, auth_token, params):
     """Given an auth token, generates filenames and sizes for SYM files.
 
-    :param auth_token: auth token for symbols.mozilla.org
-    :param url: url for file uploads
-    :param start_page: the page of files to start with
+    :param baseurl: a base url
+    :param auth_token: an auth token
+    :param params: a dict of parameters to pass by querystring.
 
     :returns: generator of (key, size) typles
 
     """
-    sym_files = []
-    page = start_page
-    params = {"page": start_page}
+    url = urljoin(baseurl.rstrip("/"), "/api/uploads/files")
     headers = {"auth-token": auth_token, "User-Agent": "tecken-systemtests"}
+    params = params or {}
+
+    sym_files = []
+    page = params.get("page", 1)
 
     while True:
-        if sym_files:
-            yield sym_files.pop(0)
```

Comment:
I reworked this loop to use a `yield from ...` so it doesn't have multiple modes.